storage/aptosdb/src/lib.rs

```diff
@@ -198,27 +203,66 @@ fn error_if_too_many_requested(num_requested: u64, max_allowed: u64) -> Result<(
     }
 }
 
-fn update_rocksdb_properties(ledger_rocksdb: &DB, state_merkle_db: &StateMerkleDb) -> Result<()> {
+fn update_rocksdb_properties(
+    ledger_db: &LedgerDb,
+    state_merkle_db: &StateMerkleDb,
+    state_kv_db: &StateKvDb,
+) -> Result<()> {
     let _timer = OTHER_TIMERS_SECONDS
         .with_label_values(&["update_rocksdb_properties"])
         .start_timer();
-    for cf_name in ledger_db_column_families() {
+
+    let set_property_fn = |cf_name: &str, db: &DB| -> Result<()> {
         for (rockdb_property_name, aptos_rocksdb_property_name) in &*ROCKSDB_PROPERTY_MAP {
             ROCKSDB_PROPERTIES
                 .with_label_values(&[cf_name, aptos_rocksdb_property_name])
-                .set(ledger_rocksdb.get_property(cf_name, rockdb_property_name)? as i64);
+                .set(db.get_property(cf_name, rockdb_property_name)? as i64);
+        }
+        Ok(())
+    };
+
+    let split_ledger = state_kv_db.enabled_sharding();
+
+    if split_ledger {
+        for cf in ledger_metadata_db_column_families() {
+            set_property_fn(cf, ledger_db.metadata_db())?;
+        }
+
+        for cf in write_set_db_column_families() {
+            set_property_fn(cf, ledger_db.write_set_db())?;
+        }
+
+        for cf in transaction_info_db_column_families() {
+            set_property_fn(cf, ledger_db.transaction_info_db())?;
+        }
+
+        for cf in transaction_db_column_families() {
+            set_property_fn(cf, ledger_db.transaction_db())?;
+        }
+
+        for cf in event_db_column_families() {
+            set_property_fn(cf, ledger_db.event_db())?;
+        }
+
+        for cf in transaction_accumulator_db_column_families() {
+            set_property_fn(cf, ledger_db.transaction_accumulator_db())?;
+        }
+
+        for cf in state_kv_db_column_families() {
+            set_property_fn(cf, state_kv_db.metadata_db())?;
+        }
+    } else {
+        for cf in ledger_db_column_families() {
+            set_property_fn(cf, ledger_db.metadata_db())?;
         }
     }
+
     for cf_name in state_merkle_db_column_families() {
-        for (rockdb_property_name, aptos_rocksdb_property_name) in &*ROCKSDB_PROPERTY_MAP {
-            // TODO(grao): Support sharding here.
-            ROCKSDB_PROPERTIES
-                .with_label_values(&[cf_name, aptos_rocksdb_property_name])
-                .set(
-                    state_merkle_db
-                        .metadata_db()
-                        .get_property(cf_name, rockdb_property_name)? as i64,
-                );
+        set_property_fn(cf_name, state_merkle_db.metadata_db())?;
+        if state_merkle_db.sharding_enabled() {
+            for shard in 0..NUM_STATE_SHARDS {
+                set_property_fn(cf_name, state_merkle_db.db_shard(shard as u8))?;
```

Comment:
can we make a fake cf_name that includes the shard name? (similar concern seems not addressed for the kv cfs either?